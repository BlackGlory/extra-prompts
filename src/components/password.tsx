import { Box, Text as InkText } from 'ink'
import { PasswordInput } from '@components/password-input.js'

interface IPasswordProps {
  message: string
  onSubmit(value: string): void
}

export function Password(props: IPasswordProps) {
  const { message, onSubmit } = props

  return (
    <Box>
      <InkText>{message}: </InkText>
      <PasswordInput onSubmit={onSubmit} />
    </Box>
  )
}
