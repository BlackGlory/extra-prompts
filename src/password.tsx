import React from 'react'
import { Password } from '@components/password'
import { render } from 'ink'

export function password(message: string): Promise<string> {
  return new Promise(resolve => {
    const { unmount } = render(
      <Password
        message={message}
        onSubmit={value => {
          unmount()
          resolve(value)
        }}
      />
    )
  })
}
