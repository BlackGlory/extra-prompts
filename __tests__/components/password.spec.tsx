import { Password } from '@components/password.js'
import { render } from 'ink-testing-library'
import { delay } from 'extra-promise'

describe('Password', () => {
  test('render', () => {
    const fn = vi.fn()
    const { lastFrame } = render(
      <Password
        message='message'
        onSubmit={fn}
      />
    )

    expect(lastFrame()).toBe('message:')
    expect(fn).toBeCalledTimes(0)
  })

  test('onSubmit', async () => {
    const fn = vi.fn()
    const { stdin, lastFrame } = render(
      <Password
        message='message'
        onSubmit={fn}
      />
    )

    await delay(100)
    stdin.write('hello')
    await delay(100)
    stdin.write('\r')
    await delay(100)

    expect(lastFrame()).toBe('message: *****')
    expect(fn).toBeCalledTimes(1)
    expect(fn).toBeCalledWith('hello')
  })
})
