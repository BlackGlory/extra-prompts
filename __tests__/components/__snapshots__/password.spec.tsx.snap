// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Password > onSubmit 1`] = `"message: *****"`;

exports[`Password > render 1`] = `"message:"`;
